src/main/java/executor/service/service/impl/JSONFileScenarioProvider.java

```diff
@@ -0,0 +1,38 @@
+package executor.service.service.impl;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import executor.service.model.Scenario;
+import executor.service.service.ScenarioProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+public class JSONFileScenarioProvider implements ScenarioProvider {
+
+    private static final Logger log = LoggerFactory.getLogger(ScenarioProvider.class);
+    private static final String FILE_NAME = "/scenarios.json";
+
+    @Override
+    public List<Scenario> readScenarios() {
+        List<Scenario> scenarios = null;
+        try (InputStream inputStream = getClass().getResourceAsStream(FILE_NAME);
+             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+            scenarios = parseScenariosFromJson(reader);
+        } catch (IOException e) {
+            log.error("Cannot read file {}. Reason: {}", FILE_NAME, e.getMessage(), e);
+        }
+        return scenarios;
+    }
+
+    private List<Scenario> parseScenariosFromJson(BufferedReader reader) {
+        Type listType = new TypeToken<List<Scenario>>() {}.getType();
+        return new Gson().fromJson(reader, listType);
+    }
```

Comment:
This approach violates the Single Responsibility Principle. Class read from file and class parse information.

Reply:
1. Use "new Properties Config.getProperties(String fileName)" to get "InputStream". 
2. In the class "JSONFileScenarioProvider.class" in the method "List<Scenario> parseScenariosFromJson(InputStream stream)" wrap "BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))" and use for parsing.

Reply:
> This approach violates the Single Responsibility Principle. Class read from file and class parse information.

No, it don't. Interface itself is used for providing scenario list. In this implementation this task is divided into 2 parts: read and parse.

Reply:
> 1. Use "new Properties Config.getProperties(String fileName)" to get "InputStream".
> 2. In the class "JSONFileScenarioProvider.class" in the method "List parseScenariosFromJson(InputStream stream)" wrap "BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))" and use for parsing.

1. Not sure whether it will be useful. File provider is just simple case for demonstration. Again remember YAGNI.

Reply:
Here we are talking about the fact that code reuse is good.